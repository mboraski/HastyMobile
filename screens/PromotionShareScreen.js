// Third Party Imports
import React, { Component } from 'react';
import {
    StyleSheet,
    View
} from 'react-native';
import { connect } from 'react-redux';

// Relative Imports
import ShareButton from '../components/ShareButton';
import Color from '../constants/Color';
import { emY } from '../utils/em';
import instagramIcon from '../assets/icons/instagram_white.png';
import twitterIcon from '../assets/icons/twitter_white.png';
import facebookIcon from '../assets/icons/facebook_white.png';
import pinterestIcon from '../assets/icons/pinterest_white.png';
import closeIcon from '../assets/icons/close.png';


class PromotionShareScreen extends Component {
    static navigationOptions = {
        title: 'Promotion Share',
        header: null 
    };

    close = () => {
        this.props.navigation.goBack();
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.row}>
                    <ShareButton style={styles.buton} source={instagramIcon} onPress={() => {}} />
                    <ShareButton style={styles.buton} source={twitterIcon} onPress={() => {}} />
                </View>
                <View style={styles.row}>
                    <ShareButton style={styles.buton} source={facebookIcon} onPress={() => {}} />
                    <ShareButton style={styles.buton} source={pinterestIcon} onPress={() => {}} />
                </View>
                <View style={styles.bottomSection}>
                    <ShareButton style={styles.closeButton} source={closeIcon} onPress={this.close} />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: Color.WHITE
    },
    row: {
        flexDirection: 'row',
        justifyContent: 'center',
        marginBottom: emY(1.25)
    },
    buton: {
        backgroundColor: Color.GREY_400,
        marginHorizontal: 10
    },
    bottomSection: {
        position: 'absolute',
        bottom: emY(3.86),
        left: 0,
        right: 0,
        alignItems: 'center'
    },
    closeButton: {
        borderColor: Color.BLACK,
        borderWidth: 1
    }
});

const mapStateToProps = state => ({
    header: state.header
});

export default connect(mapStateToProps, {})(PromotionShareScreen);
