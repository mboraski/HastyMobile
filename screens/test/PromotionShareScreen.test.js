import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import PromotionShareScreen from '../PromotionShareScreen';

const initialState = {};

describe('PromotionShareScreen', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<PromotionShareScreen />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
