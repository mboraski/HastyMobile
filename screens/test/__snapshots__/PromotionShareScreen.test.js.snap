// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionShareScreen renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
    }
>
    <View
        style={
            Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
              }
        }
    >
        <Connect(ShareButton)
            source={1}
            style={
                Object {
                    "backgroundColor": "#b2b2b2",
                    "marginHorizontal": 10,
                  }
            }
        />
        <Connect(ShareButton)
            source={1}
            style={
                Object {
                    "backgroundColor": "#b2b2b2",
                    "marginHorizontal": 10,
                  }
            }
        />
    </View>
    <View
        style={
            Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
              }
        }
    >
        <Connect(ShareButton)
            source={1}
            style={
                Object {
                    "backgroundColor": "#b2b2b2",
                    "marginHorizontal": 10,
                  }
            }
        />
        <Connect(ShareButton)
            source={1}
            style={
                Object {
                    "backgroundColor": "#b2b2b2",
                    "marginHorizontal": 10,
                  }
            }
        />
    </View>
    <View
        style={
            Object {
                "alignItems": "center",
                "bottom": 61.76,
                "left": 0,
                "position": "absolute",
                "right": 0,
              }
        }
    >
        <Connect(ShareButton)
            source={1}
            style={
                Object {
                    "borderColor": "#000000",
                    "borderWidth": 1,
                  }
            }
        />
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
      }
>
      <View
            style={
                  Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginBottom": 20,
                      }
            }
      >
            <Connect(ShareButton)
                  source={1}
                  style={
                        Object {
                              "backgroundColor": "#b2b2b2",
                              "marginHorizontal": 10,
                            }
                  }
            />
            <Connect(ShareButton)
                  source={1}
                  style={
                        Object {
                              "backgroundColor": "#b2b2b2",
                              "marginHorizontal": 10,
                            }
                  }
            />
      </View>
      <View
            style={
                  Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginBottom": 20,
                      }
            }
      >
            <Connect(ShareButton)
                  source={1}
                  style={
                        Object {
                              "backgroundColor": "#b2b2b2",
                              "marginHorizontal": 10,
                            }
                  }
            />
            <Connect(ShareButton)
                  source={1}
                  style={
                        Object {
                              "backgroundColor": "#b2b2b2",
                              "marginHorizontal": 10,
                            }
                  }
            />
      </View>
      <View
            style={
                  Object {
                        "alignItems": "center",
                        "bottom": 61.76,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                      }
            }
      >
            <Connect(ShareButton)
                  source={1}
                  style={
                        Object {
                              "borderColor": "#000000",
                              "borderWidth": 1,
                            }
                  }
            />
      </View>
</View>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <PromotionShareScreen />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PromotionShareScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#ffffff",
                              "flex": 1,
                              "flexDirection": "column",
                              "justifyContent": "center",
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "marginBottom": 20,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
          </View>
          <View
                    style={
                              Object {
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "marginBottom": 20,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
          </View>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "bottom": 61.76,
                                        "left": 0,
                                        "position": "absolute",
                                        "right": 0,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "borderColor": "#000000",
                                                  "borderWidth": 1,
                                                }
                              }
                    />
          </View>
</View>,
        "_debugID": 4,
        "_renderedOutput": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#ffffff",
                              "flex": 1,
                              "flexDirection": "column",
                              "justifyContent": "center",
                            }
          }
>
          <View
                    style={
                              Object {
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "marginBottom": 20,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
          </View>
          <View
                    style={
                              Object {
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "marginBottom": 20,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "backgroundColor": "#b2b2b2",
                                                  "marginHorizontal": 10,
                                                }
                              }
                    />
          </View>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "bottom": 61.76,
                                        "left": 0,
                                        "position": "absolute",
                                        "right": 0,
                                      }
                    }
          >
                    <Connect(ShareButton)
                              source={1}
                              style={
                                        Object {
                                                  "borderColor": "#000000",
                                                  "borderWidth": 1,
                                                }
                              }
                    />
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PromotionShareScreen />,
}
`;
